ctapipe/image/modifications.py

```diff
@@ -0,0 +1,174 @@
+):
+    """
+    Create a new image with values smeared across the
+    neighbor pixels specified by `indices` and `indptr`.
+    These are what defines the sparse neighbor matrix
+    and are available as attributes from the neighbor matrix.
+    The amount of charge that is distributed away from a given
+    pixel is drawn from a poissonian distribution.
+    The distribution of this charge among the neighboring
+    pixels follows a multinomial.
+    Pixels at the camera edge lose charge this way.
+    No geometry is available in this function due to the numba optimization,
+    so the indices, indptr and smear_probabilities have to match
+    to get sensible results.
+
+    Parameters:
+    -----------
+    image: ndarray
+        1d array of the pixel charge values
+    fraction: float
+        fraction of charge that will be distributed among neighbors (modulo poissonian)
+    indices: ndarray[int]
+        CSR format index array of the neighbor matrix
+    indptr: ndarray[int]
+        CSR format index pointer array of the neighbor matrix
+    smear_probabilities: ndarray[float]
+        shape: (n_neighbors, )
+        A priori distribution of the charge amongst neighbors.
+        In most cases probably of the form np.full(n_neighbors, 1/n_neighbors)
+    seed: int
+        Random seed for the numpy rng.
+        Because this is numba optimized, a rng instance can not be used here
+
+    Returns:
+    --------
+    new_image: ndarray
+        1d array with smeared values
+    """
+    new_image = image.copy()
+    if seed is not None:
+        np.random.seed(seed)
+
+    for pixel in range(len(image)):
+
+        if image[pixel] <= 0:
+            continue
+
+        to_smear = np.random.poisson(image[pixel] * fraction)
+        if to_smear == 0:
+            continue
+
+        # remove light from current pixel
+        new_image[pixel] -= to_smear
+
+        # add light to neighbor pixels
+        neighbors = indices[indptr[pixel] : indptr[pixel + 1]]
+        n_neighbors = len(neighbors)
+
+        # all neighbors are equally likely to receive the charge
+        # we always distribute the charge into 6 neighbors, so that charge
+        # on the edges of the camera is lost
+        neighbor_charges = np.random.multinomial(to_smear, smear_probabilities)
+
+        for n in range(n_neighbors):
+            neighbor = neighbors[n]
+            new_image[neighbor] += neighbor_charges[n]
+
+    return new_image
+
+
+class ImageModifier(TelescopeComponent):
+    """
+    Abstract class for configurable image modifying algorithms. Use
+    ``ImageModifier.from_name()`` to construct an instance of a particular algorithm
+    """
+
+    @abstractmethod
+    def __call__(self, tel_id: int, image: np.ndarray) -> np.ndarray:
+        """
+        Modifies a given image
+
+        Returns
+        -------
+        np.ndarray
+            modified image
+        """
+
+
+class NSBNoiseAdder(ImageModifier):
```

Comment:
The class should be called differently now. Probably this should just be `ImageModifier`.

Reply:
No base class anymore?

Reply:
Yes.